packages/bruno-app/src/components/CollectionSettings/Auth/OAuth2/CredentialsPreview/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { clearOauth2Cache, readOauth2CachedCredentials } from 'utils/network';
+import { sendCollectionOauth2Request } from 'providers/ReduxStore/slices/collections/actions';
+import toast from 'react-hot-toast';
+import { useDispatch } from 'react-redux';
+
+const CredentialsPreview = ({ collection }) => {
+  const oauth2CredentialsAreaRef = React.createRef();
+  const [oauth2Credentials, setOauth2Credentials] = useState({});
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    oauth2CredentialsAreaRef.current.value = oauth2Credentials;
+    readOauth2CachedCredentials(collection.uid).then((credentials) => setOauth2Credentials(credentials));
+  }, [oauth2CredentialsAreaRef]);
+
+  const handleRun = async () => {
+    dispatch(sendCollectionOauth2Request(collection.uid));
+  };
+
+  const handleClearCache = (e) => {
+    clearOauth2Cache(collection?.uid)
+      .then(() => {
+        readOauth2CachedCredentials(collection.uid).then((credentials) => {
+          setOauth2Credentials(credentials);
+          toast.success('cleared cache successfully');
+        });
+      })
+      .catch((err) => {
+        toast.error(err.message);
+      });
+  };
+
+  return (
+    <div className="flex flex-col w-full gap-1 mt-4">
+      <div className="flex flex-row gap-4">
+        <button onClick={handleRun} className="submit btn btn-sm btn-secondary w-fit">
+          Get Access Token
+        </button>
+        <button onClick={handleClearCache} className="submit btn btn-sm btn-secondary w-fit">
+          Clear Cache
+        </button>
+      </div>
+      <div ref={oauth2CredentialsAreaRef}>
+        <label className="flex flex-row w-full mt-2 gap-2">Cached OAuth2 Credentials:</label>
+        {Object.entries(oauth2Credentials).map(([key, value]) => (
+          <div key={key}>
+            <label className="text-xs">{key}</label>
+            <textarea className="w-full h-24 p-2 text-xs border rounded" value={value} readOnly />
```

Comment:
the background/text color needs to be updated for dark mode

What it looks like now:
<img width="788" alt="image" src="https://github.com/usebruno/bruno/assets/18120837/5f7241ef-bd41-481f-8a22-69ceb1eff63e">

Reply:
It now looks more like this:
![Screenshot from 2024-05-05 19-07-51](https://github.com/usebruno/bruno/assets/2108093/b687288a-d69e-4c0b-8cc5-108c47b0cfc4)